stubs/aiofiles/aiofiles/tempfile/__init__.pyi

```diff
@@ -137,6 +78,133 @@ def TemporaryFile(
     executor: Incomplete | None = None,
 ) -> AiofilesContextManager[None, None, AsyncBufferedIOBase]: ...
 
+# 3.12 added `delete_on_close`
+if sys.version_info >= (3, 12):
+    # Text mode: always returns AsyncTextIOWrapper
+    @overload
+    def NamedTemporaryFile(
+        mode: OpenTextMode,
+        buffering: int = -1,
+        encoding: str | None = None,
+        newline: str | None = None,
+        suffix: AnyStr | None = None,
+        prefix: AnyStr | None = None,
+        dir: StrOrBytesPath | None = None,
+        delete: bool = True,
+        delete_on_close: bool = True,
```

Comment:
Should this be keyword-only as in `tempfile.pyi`?

Reply:
[Taking a look](https://github.com/Tinche/aiofiles/blob/v23.2.1/src/aiofiles/tempfile/__init__.py#L24-L61), the interface differs in that regard, and keyword arguments shouldn't be enforced on any `NamedTemporaryFile` overloads.